core-rust-bridge/src/main/java/com/radixdlt/statecomputer/commit/CommitRequest.java

```diff
@@ -75,7 +76,9 @@
 public record CommitRequest(
     List<RawLedgerTransaction> transactions,
     LedgerProof proof,
-    Option<byte[]> postCommitVertexStoreBytes) {
+    Option<byte[]> postCommitVertexStoreBytes,
+    // for metrics calculation only
+    Option<ComponentAddress> selfValidatorAddress) {
```

Comment:
_(note from author)_

> Why is this a part of `CommitRequest` and not a part of `StateManager`'s initial configuration?

Because injecting `SelfValidatorInfo` to a `NodeRustEnvironment` introduces a circular dependency (the construction of `SelfValidatorInfo` requires `REv2LedgerInitializerToken`, which cannot be constructed without having `NodeRustEnvironment`).

I am ashamed of how we use Guice - still, if you see a way of improving this very line with a quick-win (i.e. without refactoring our entire DI), I am open to suggestions.